### ISSUE TITLE
API: dask.DataFrame.to* return values

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask#1622]
When writing a `dask.DataFrame` to multiple files on disk with globs, it could be useful to return the file paths written to. For example,  

``` python
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame({"A": range(10)})
a = dd.from_pandas(df, npartitions=2)

a.to_csv("foo*.csv")
# return value = ['foo0.csv', foo1.csv']

a.to_hdf("foo*.hdf5", "key")
# return value = ['foo0.hdf5', foo1.hdf5']
```

Currently `to_csv` returns None (as in pandas) and `to_hdf` returns a tuple of Nones (I assume equal to the number of partitions), which differs from pandas, which again has no return value.

My use-case is having one function pull from a database and write to disk. Currently I return `glob.glob('foo*.hdf5')`, but then I need to be careful that nothing else matches that pattern.

I _think_ this is a case where it's OK for dask to differ from the pandas API. Though I could be missing some corner cases.

**Comment by mrocklin** [comment]
This seems like a good idea to me

**Comment by mrocklin** [comment]
Closed by #2655
